File: data/src/main/java/com/fishbrain/android/domain/Species.java
package com.fishbrain.android.domain;

import com.google.gson.annotations.SerializedName;

/**
 * This is the POJO for species
 *
 * Created by dimitris.lachanas on 22/08/15.
 */
public class Species {
    @SerializedName("id")
    private String mId;

    @SerializedName("name")
    private String mName;

    @SerializedName("species")
    private String mSpecies;

    @SerializedName("status")
    private String mStatus;

    @SerializedName("image")
    private String mImage;

    @SerializedName("photo")
    private Photo mPhoto;

    public Species(String mId, String mName, String mSpecies, String mStatus, String mImage, Photo photo) {
        this.mId = mId;
        this.mName = mName;
        this.mSpecies = mSpecies;
        this.mStatus = mStatus;
        this.mImage = mImage;
        this.mPhoto = photo;
    }

    public String getId() {
        return mId;
    }

    public String getName() {
        return mName;
    }

    public String getSpecies() {
        return mSpecies;
    }

    public String getStatus() {
        return mStatus;
    }

    public String getImage() {
        return mImage;
    }

    public Photo getPhoto() {
        return mPhoto;
    }
}

File: app/src/main/java/com/fishbrain/assignment/data/domain/Species.kt
package com.fishbrain.assignment.data.domain

import com.google.gson.annotations.SerializedName

/**
 * This is the POJO for species
 *
 * Created by dimitris.lachanas on 22/08/15.
 */
data class Species(
        @field:SerializedName("id") val id: String,
        @field:SerializedName("name") val name: String,
        @field:SerializedName("species") val species: String,
        @field:SerializedName("status") val status: String,
        @field:SerializedName("image") val image: String,
        @field:SerializedName("photo") val photo: Photo
)
